source/code/plugin/in_containerinventory.rb

```diff
@@ -0,0 +1,234 @@
+#!/usr/local/bin/ruby
+# frozen_string_literal: true
+
+module Fluent
+
+  class Container_Inventory_Input < Input
+    Plugin.register_input('containerinventory', self)
+
+    def initialize
+      super
+      require 'json'
+      require_relative 'DockerApiClient'
+      require_relative 'ContainerInventoryState'
+      require_relative 'omslog'
+    end
+
+    config_param :run_interval, :time, :default => '1m'
+    config_param :tag, :string, :default => "oms.containerinsights.ContainerInventory"
+  
+    def configure (conf)
+      super
+    end
+
+    def start
+      if @run_interval
+        @finished = false
+        @condition = ConditionVariable.new
+        @mutex = Mutex.new
+        @thread = Thread.new(&method(:run_periodic))
+      end
+    end
+
+    def shutdown
+      if @run_interval
+        @mutex.synchronize {
+          @finished = true
+          @condition.signal
+        }
+        @thread.join
+      end
+    end
+
+    def obtainContainerConfig(instance, container)
+      begin
+        configValue = container['Config']
+        if !configValue.nil?
+          instance['ContainerHostname'] = configValue['Hostname']
+
+          envValue = configValue['Env']
+          envValueString = (envValue.nil?) ? "" : envValue.to_s
+          instance['EnvironmentVar'] = envValueString
+
+          cmdValue = configValue['Cmd']
+          cmdValueString = (cmdValue.nil?) ? "" : cmdValue.to_s
+          instance['Command'] = cmdValueString
+
+          instance['ComposeGroup'] = ""
```

Comment:
where is this used?

Reply:
This is to set the ComposeGroup value in the container inventory